src/vl53_main.c

```diff
@@ -0,0 +1,270 @@
+/************************************
+ *       VL53 Main source file      *
+ ************************************/
+
+// Includes
+#include "vl53_main.h"
+#include "hardware_functions.h"
+#include "error_handling.h"
+
+//Global variables
+extern uint16_t PeopleCount;
+extern uint8_t vl53_Counting;
+
+// Programmable I2C address of the VL53L1X sensor
+uint16_t VL53_I2C_Address = (0x29 << 1); // 7 bit, MSB
+
+// Local variables
+static uint16_t Distance = 0, last_PeopleCount = 0;
+static uint8_t status = 0, sensorState = 0, RangeStatus = 0, dataReady = 0;
+
+int VL53_Setup()
+{
+    // Wait for sensor to boot
+    while (sensorState == 0)
+    {
+        status = VL53L1X_BootState(VL53_I2C_Address, &sensorState);
+        HAL_Delay(2);
+    }
+    sensorState = 0;
+
+// Print process, debug purposes
+#ifdef debug
+    UART_PutStr((uint8_t *)"Sensor booted\r\n");
+#endif
+
+    // Set desired register settings
+    status = VL53L1X_SensorInit(VL53_I2C_Address);

```

Comment:
waarom tel je dit bij status op?

Reply:
om te kijken of er ergens tussen die functies iets fout gaat. Zit in de library ingebakken